qbft/commit.go

```diff
@@ -120,43 +122,52 @@ func baseCommitValidationIgnoreSignature(
 
 func baseCommitValidationVerifySignature(
 	config IConfig,
-	signedCommit *SignedMessage,
+	signedCommit *types.SignedSSVMessage,
 	height Height,
-	operators []*types.Operator,
-) error {
+	operators []*types.CommitteeMember) error {
 
 	if err := baseCommitValidationIgnoreSignature(signedCommit, height, operators); err != nil {
 		return err
 	}
 
 	// verify signature
-	if err := signedCommit.Signature.VerifyByOperators(signedCommit, config.GetSignatureDomainType(), types.QBFTSignatureType, operators); err != nil {
+	if err := config.GetSignatureVerifier().Verify(signedCommit, operators); err != nil {
 		return errors.Wrap(err, "msg signature invalid")
 	}
 
 	return nil
 }
 
 func validateCommit(
-	signedCommit *SignedMessage,
+	signedCommit *types.SignedSSVMessage,
 	height Height,
 	round Round,
-	proposedMsg *SignedMessage,
-	operators []*types.Operator,
+	proposedSignedMsg *types.SignedSSVMessage,
+	operators []*types.CommitteeMember,
 ) error {
 	if err := baseCommitValidationIgnoreSignature(signedCommit, height, operators); err != nil {
 		return err
 	}
 
-	if len(signedCommit.Signers) != 1 {
+	if len(signedCommit.GetOperatorIDs()) != 1 {
 		return errors.New("msg allows 1 signer")
 	}
 
-	if signedCommit.Message.Round != round {
+	msg, err := DecodeMessage(signedCommit.SSVMessage.Data)
+	if err != nil {
+		return err
+	}
+
+	if msg.Round != round {
 		return errors.New("wrong msg round")
 	}
 
-	if !bytes.Equal(proposedMsg.Message.Root[:], signedCommit.Message.Root[:]) {
+	proposedMsg, err := DecodeMessage(proposedSignedMsg.SSVMessage.Data)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(proposedMsg.Root[:], msg.Root[:]) {
 		return errors.New("proposed data mistmatch")
```

Comment:
Not part of this PR, but we might fix it now as we're touching this function. If you decide to fix it, please search for this word in other files too

Reply:
```suggestion
		return errors.New("proposed data mismatch")
```